### PR TITLE
Start a dependency graph section in the specs

### PR DESCRIPTION
**Pull request by starksm64**
This adds the web profile dependencies using asciidoctor-diagram and the graphviz format.

Signed-off-by: starksm64 <starksm64@gmail.com>

### CODE REVIEW

File: specification/src/main/asciidoc/web-dependencies.adoc [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+digraph EE10WebProfile {
+
+# Map a short node id to the label with version
+wp [label="jakarta-web-api"];
+servlet [label="servlet:6.0.0"];
+jsp [label="servlet.jsp:3.1.0"];
+el [label="el:5.0.0"];
+jstl [label="servlet.jsp.jstl:2.0.0"];
+jaxb [label="xml.bind:3.0.0"];
+faces [label="faces:3.0.0"];
```
Comment (ivargrimstad) [inline specification/src/main/asciidoc/web-dependencies.adoc:14]: faces 4.0.0

File: specification/src/main/asciidoc/web-dependencies.adoc [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+websocket [label="websocket:2.0.0"];
+jsonp [label="jsonp:2.1.0"];
+jsonb [label="jsonb:2.0.0"];
+annotations [label="annotations:2.0.0"];
+ejb [label="ejb:4.0.0"];
+jta [label="jta:2.0.0"];
+jpa [label="jpa:3.0.0"];
+bv [label="bv:3.0.1"];
+interceptors [label="interceptors:2.0.0"];
+cdi [label="cdi:3.0.0"];
```
Comment (ivargrimstad) [inline specification/src/main/asciidoc/web-dependencies.adoc:25]: CDI 4.0.0

File: specification/src/main/asciidoc/web-dependencies.adoc [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+di [label="di:2.0.1"];
+auth [label="auth:2.0.0"];
+jacc [label="jacc:2.0.0"];
+xml_bind [label="xml.bind:3.0.1"];
+activation [label="activation:2.0.1"];
+
```
Comment (ivargrimstad) [inline specification/src/main/asciidoc/web-dependencies.adoc:31]: Acc Concurrency 3.0.0

Comment (ivargrimstad) [inline specification/src/main/asciidoc/web-dependencies.adoc:31]: Add Security 3.0.0

File: specification/src/main/asciidoc/web-dependencies.adoc [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+jaxb [label="xml.bind:3.0.0"];
+faces [label="faces:3.0.0"];
+jaxrs [label="jaxrsr:3.1.0"]
+websocket [label="websocket:2.0.0"];
+jsonp [label="jsonp:2.1.0"];
+jsonb [label="jsonb:2.0.0"];
+annotations [label="annotations:2.0.0"];
+ejb [label="ejb:4.0.0"];
+jta [label="jta:2.0.0"];
+jpa [label="jpa:3.0.0"];
```
Comment (ivargrimstad) [inline specification/src/main/asciidoc/web-dependencies.adoc:22]: JPA 3.1.0

File: specification/src/main/asciidoc/web-dependencies.adoc [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+servlet [label="servlet:6.0.0"];
+jsp [label="servlet.jsp:3.1.0"];
+el [label="el:5.0.0"];
+jstl [label="servlet.jsp.jstl:2.0.0"];
+jaxb [label="xml.bind:3.0.0"];
+faces [label="faces:3.0.0"];
+jaxrs [label="jaxrsr:3.1.0"]
+websocket [label="websocket:2.0.0"];
+jsonp [label="jsonp:2.1.0"];
+jsonb [label="jsonb:2.0.0"];
```
Comment (ivargrimstad) [inline specification/src/main/asciidoc/web-dependencies.adoc:18]: JSON-B 3.0.0

File: specification/src/main/asciidoc/web-dependencies.adoc [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+# Map a short node id to the label with version
+wp [label="jakarta-web-api"];
+servlet [label="servlet:6.0.0"];
+jsp [label="servlet.jsp:3.1.0"];
+el [label="el:5.0.0"];
+jstl [label="servlet.jsp.jstl:2.0.0"];
+jaxb [label="xml.bind:3.0.0"];
+faces [label="faces:3.0.0"];
+jaxrs [label="jaxrsr:3.1.0"]
+websocket [label="websocket:2.0.0"];
```
Comment (ivargrimstad) [inline specification/src/main/asciidoc/web-dependencies.adoc:16]: WebSocket 2.1.0

File: specification/src/main/asciidoc/web-dependencies.adoc [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+jsonp [label="jsonp:2.1.0"];
+jsonb [label="jsonb:2.0.0"];
+annotations [label="annotations:2.0.0"];
```
Comment (ivargrimstad) [inline specification/src/main/asciidoc/web-dependencies.adoc:19]: Annotations 2.1.0

File: specification/src/main/asciidoc/web-dependencies.adoc [hunk 8]
```diff
@@ -0,0 +1,102 @@
...
+[graphviz]
+-----------------------------------------------------------
+digraph EE10WebProfile {
+
+# Map a short node id to the label with version
+wp [label="jakarta-web-api"];
+servlet [label="servlet:6.0.0"];
+jsp [label="servlet.jsp:3.1.0"];
+el [label="el:5.0.0"];
+jstl [label="servlet.jsp.jstl:2.0.0"];
```
Comment (ivargrimstad) [inline specification/src/main/asciidoc/web-dependencies.adoc:12]: JSTL 3.0.0

**Review by ivargrimstad — COMMENTED** [review]
Looks great!
Made a couple of comments regarding some of the version numbers. I think I caught them all.

File: specification/src/main/asciidoc/web-dependencies.adoc (hunk 3 above)
Comment (ivargrimstad) [inline specification/src/main/asciidoc/web-dependencies.adoc:31]: Add Managed Beans 2.0

Comment (ivargrimstad) [inline specification/src/main/asciidoc/web-dependencies.adoc:31]: Add Debugging Support 2.0

**Comment by ivargrimstad** [comment]
I may have been a little too eager with the version numbers if the diagram is meant to reflect the current staged versions, and not the intended EE 10 versions...

**Comment by starksm64** [comment]
Right, these only have  been updated to what has been staged or released as determined by what can be used in the jarkarta.platform api project. As new specs are staged this can be updated.

**Comment by starksm64** [comment]
There are early releases of some api that could also be updated, CDI for example.

**Review by ivargrimstad — APPROVED** [review]

**Pull request closed without merging**